table/block_based/block_based_table_reader_sync_and_async.h

```diff
@@ -402,6 +402,7 @@ DEFINE_SYNC_AND_ASYNC(void, BlockBasedTable::MultiGet)
         BCI block_cache{rep_->table_options.block_cache.get()};
         std::array<BCI::TypedAsyncLookupHandle, MultiGetContext::MAX_BATCH_SIZE>
             async_handles;
+        BlockCreateContext create_ctx = rep_->create_context;
```

Comment:
Unfortunate copy. Are we sure it is necessary?

Reply:
The reason for using a local copy is to set the compression dictionary. Not sure how to avoid it, unless we add a 2nd argument that can be passed to the creation callback. Or maybe I can add a wrapper around ```BlockCreateContext``` with pointers and pass that instead.